object/slicer/slicer.go

```diff
@@ -33,59 +37,30 @@ type ObjectWriter interface {
 	// Signer is required and must not be nil. The operation is executed on behalf of
 	// the account corresponding to the specified Signer, which is taken into account, in particular, for access control.
 	InitDataStream(header object.Object, signer neofscrypto.Signer) (dataStream io.Writer, err error)
+
+	// SetSession pass session object to writer. It will be used for writing object routine.
+	// This method must be called before InitDataStream.
+	SetSession(sess session.Object)
```

Comment:
requirements are pretty strict, maybe it'd be clearer to pass `session.Object` parameter into `InitDataStream`?

Reply:
I may suggest an alternative solution:
- remove `SetSession` from the interface
- update `func NewDataSlicer(ctx context.Context, cli *Client) (*slicer.Slicer, error) {` 
to `func NewDataSlicer(ctx context.Context, cli *Client, sess *session.Object) (*slicer.Slicer, error) {`.
In this case we hide session inside `objectWriter` and don't need to set it outside
- make objectWriter public as a default implementation for the interface. It's constructor even may have `session.Object` as a parameter

Reply:
> requirements are pretty strict

we thought that we need a way to be able to create a slicer once and slice more than one object and send parts to more than one node (so more than one session will be attached). we can discuss it

Reply:
Do I apply my suggestions?

Reply:
how about provide both `client.NewDataSlicer` and `client.NewDataSlicerWithSession`?

Reply:
Right. The option with two constructors for `ObjectWriter` also looks fine to me and it will help us to remove `SetSession(sess session.Object)` from the interface

Reply:
@cthulhu-rider, how that allows us to send the parts to different nodes with only one `Slicer` (the problem/wish i described above)?

Reply:
@carpawell

> send the parts to different nodes

we can do it with single session token, can't we? Afaik we don't deal with dynamic sessions when we talk about `Slicer`. The app just takes user's session token and attach it to any node it send objects to

so pls clarify ur concern about `different nodes`, maybe i didn't get ur point

Reply:
> Afaik we don't deal with dynamic sessions when we talk about Slicer

i said smth wrong. well, the question is about reusing the `Slicer` then. do we want to be able to create `Slicer` once and handle any object from any user?